### ISSUE TITLE
New CTL programs

### ISSUE THREAD

**Issue by aforsythe** [ampas/CTL#144]
I've come across a few scenarios where having programs capable of tasks not currently supported by the built-in tools in CTL would be highly advantageous.

The first scenario involves the desire to input a scalar, vector, or array of numbers directly into CTL and observe the numerical output. At present, we utilize CTL through Matlab by creating a temporary image, processing it with ctlrender, and then importing the results back into Matlab. I believe @KelSolaar employs a similar method with `[colour-science](https://github.com/colour-science/colour/blob/5adf88b737196a94bd80a69f80276b48ee87e1c5/colour/io/ctl.py#L102)`. Ideally, a program that accepts numerical data directly might be more efficient, although I acknowledge there might be intricate details that need to be addressed.

The second scenario is the need to execute a CTL program without feeding an image into it. This functionality would be particularly beneficial for conducting unit tests, such as the ones I have outlined here: https://github.com/ampas/aces-unittest-test. Currently, my approach involves using ctlrender, even though the image data itself isn't being utilized, which seems like an inelegant solution.

I'd be interested in hearing your opinions. Thanks.

**Comment by KelSolaar** [comment]
Passing a scalar or vector and getting it transformed on stdout would be great yeah!

**Comment by michaeldsmith** [comment]
@aforsythe I'm getting a github 404 on your aces-unit-test repo page
![image](https://github.com/ampas/CTL/assets/37905569/3723a115-075a-4910-974d-1260144fe931)

**Comment by michaeldsmith** [comment]
@aforsythe @KelSolaar do you imagine this functionality should be part of a separate program independent of ctlrender or as alternate input/output path added to ctlrender, or something else ?

**Comment by aforsythe** [comment]
@michaeldsmith Sorry the repo was private.   Public now.

re: stdout ... I'm open.  I can see arguments both ways, but I think it probably makes sense to have a dedicated utility.

**Comment by KelSolaar** [comment]
> @aforsythe @KelSolaar do you imagine this functionality should be part of a separate program independent of ctlrender or as alternate input/output path added to ctlrender, or something else ?

Oh sorry, I was not clear, yes on existing `ctlrender`. I would keep things simple, it is already huge progress to see someone stepping up to update this repo so I would not want to add the extra burden of having to maintain yet another program.

**Comment by aforsythe** [comment]
I tend to agree ... thinking about it, the command line interface we'd want would probably be nearly identical to ctlrender, so why not just build on top of that.
